File: app/src/main/java/org/app/sharing/dto/UserSearchResult.java
package org.app.sharing.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserSearchResult {
    private Long   id;
    private String name;
    private String surname;
    private String lastName;
    private String photoLink;
    private Long   roomId;
    private String roomNumber;
}

File: app/src/main/java/com/coursework/pleasantroutineui/pleasantroutineui/domain/discovery/UserSearchResult.kt
package com.coursework.pleasantroutineui.domain.discovery

data class UserSearchResult(
    val id: Long,
    val name: String?,
    val surname: String?,
    val lastName: String?,
    val photoLink: String?,
    val roomId: Long?,
    val roomNumber: String?
) {
    val fullName: String
        get() = listOfNotNull(surname, name, lastName).joinToString(" ")
}